### DISCUSSION TITLE
Create a Minimal Build with Theming, IDE, and GUI for Basic Shell Configuration

### DISCUSSION THREAD

**Discussion by Jean-Monsieur** [iMAboud/iMA-Menu#4] [Ideas]
Hello,

I want to commend you on the excellent work you've done with your project. However, I have a suggestion that might enhance its usability. It would be beneficial to consider creating a version of your project that does not modify the default shell context menu application.

You've developed some impressive features and utilities that offer significant customization options. I recommend structuring your project as a monorepository. This would include your main application while keeping features such as 

-  upload/download
- YouTube/Xdownloader 
- video/image editing 
-  search 
- draw
- clean temp
- wallpaper changer 
- color picker
-  merge media 
-  StremiM 

as optional dependencies. This approach would allow users to build their own version of Shell with the excellent tools you've integrated.

Please note that these are merely suggestions. I genuinely believe your project has great potential and could enhance the user experience significantly.

Best wishes for the continued success of your project.

Have a wonderful day!

**Comment by iMAboud** [comment]
Thank you for the suggestions, however, none of the extra features are registered in shell, only the essentials which makes the software works, that are made by the Nilesoft Shell developer. the rest of the features are only scripts that work only during executions, no registry or services in background. deleting the files from installation folder and not installing the python libraries are enough to completely remove them. 

So, in a way, it's presented as optional, but more integrated in a simple 1 folder for less issues during installation. 

I guess if you want a stock NS software you can install the official release. I've kept everything that I made as optional and never to be installed as background services anyway. 

Thanks again, looking forward for your contributions!
